### ISSUE TITLE
[BUG] Index patterns not loaded after creation of the index template for all indices pattern "*"

### ISSUE THREAD

**Issue by deepak-rsystems** [opensearch-project/OpenSearch-Dashboards#2877]
**Describe the bug**

If we load an index template for all indices "*" pattern with following template settings, indices patterns are not loaded anymore and saved objects are also not visible anymore and we get "Unable find saved objects" error on screen.

`"template" : {
          "settings" : {
            "index" : {
              "routing" : {
                "allocation" : {
                  "include" : {
                    "role" : "data"
                  }
                }
              },
              "refresh_interval" : "30s"
            }
          }
        }`

We checked that problem is caused by the fact that .kibana* system indices are getting the all indices index template applied and has template setting for routing allocation.
`".kibana_1" : {
    "settings" : {
      "index" : {
        "routing" : {
          "allocation" : {
            "include" : {
              "role" : "data"
            }
          }
        },
        "refresh_interval" : "30s",
        "number_of_shards" : "1",
        "auto_expand_replicas" : "0-1",
        "provided_name" : ".kibana_1",
        "creation_date" : "1668430185144",
        "number_of_replicas" : "1",
        "uuid" : "sPO3PtfcRC2DVn_ynIcl_A",
        "version" : {
          "created" : "136247827"
        }
      }
    }
  }`

**To Reproduce**
1. Go to Saved Objects
2. See error
![saved-objects-missing](https://user-images.githubusercontent.com/60732633/201940954-ff79c93b-e0ca-4268-a556-e5ce23ad635c.png)

3. But it seems, the saved object config is present there but not visible by default:

<img width="775" alt="saved-objects-missing1" src="https://user-images.githubusercontent.com/60732633/201941237-9ca3bab7-8458-4d97-893d-841423d6138f.png">

4. And if we create an index pattern, it doesn't get saved. 
5. We also noticed that, if we restart the dashboards pod, then dashboards gui starts to work fine!!

**Expected behavior**
Opensearch dashboard functionality should not be impacted due to index allocation routing on .kibana* system indices.

**OpenSearch Version**
2.3.0

**Dashboards Version**
2.3.0

**Plugins**

No custom plugin enabled.

**Screenshots**

**Host/Environment (please complete the following information):**
 - Browser and version [e.g. 22]

**Additional context**

**Comment by IanGabes** [comment]
My team and I are is also noticing this behaviour on build 2.3.0, and 2.4.0

Additional Notes:

1) Users are ABLE to view index-patterns for the global tenant
2) Users are UNABLE to view index-patterns for private and shared tenants
3) Deleting the `routing` key from the index settings does not seem to have an effect, before/after restarting the dashboards

The user used for testing this has all permissions, on all indices and tenants. 

Here is a screenshot of the error from the Saved Objects table:

![image](https://user-images.githubusercontent.com/415953/202496932-d7bd4ae8-0bfe-4fc6-9ea1-c393c2d93703.png)

Is it possible that the Index Patterns Table is having a similar issue without returning an error to the client? They both use `title` as a sort field, without specifying the `keyword` any way that I could see.

**Comment by ananzh** [comment]
Hey @deepak-rsystems and @IanGabes, thanks for opening this issue. I am trying to reproduce this issue. Here is my reproduction steps:

* run OS and OSD
* add 3 sample data
* go to dev tools and create an index template for all index patterns using the template settings provided in the issue:
```
PUT _index_template/template_1
{
  "index_patterns": ["*"],
  "template": {
    "settings": {
      "index": {
        "routing": {
          "allocation": {
            "include": {
              "role": "data"
            }
          }
        },
        "refresh_interval": "30s"
      }
    }
  }
}
```
* go to Stack Management, click Saved Objects

I am able to see all the saved objects. Is my template correct? Is there missing steps? Thanks.

**Comment by IanGabes** [comment]
Hi @ananzh, to your repro steps i would add:

 - Have multi-tenancy enabled
 - Select the private or a shared tenant
 
 We do not notice the issue on the Global tenant. Thanks!

**Comment by kavilla** [comment]
Looks like it might be something with the security plugin like perhaps the plugin is not marking private tenants indices as system and that's why the wildcard index template is being applied.

cc: @ananzh

**Comment by joshuarrrr** [comment]
Yeah, @opensearch-project/transfer-request , I think this is more likely to be solved via https://github.com/opensearch-project/security-dashboards-plugin
